### DISCUSSION TITLE
Cannot access Github Issues/ PRs from a public Microsoft owned repository.

### DISCUSSION THREAD

**Discussion by sandrohanea** [community/community#41653] [Enterprise]
### Select Topic Area

Product Feedback

### Body

Hello,

I am part of https://github.com/microsoft organization, and I am logged on github from 2 devices (one is Microsoft owned, and the other one not).

On the one owned by Microsoft, everything is working as expected (as I am logged in with @microsoft account in my edge browser).

On the other laptop,  I am logged in with my personal account in the edge browser and whenever I tried to access public issues or PRs, it redirects me to this page:
<img width="469" alt="image" src="https://user-images.githubusercontent.com/40202887/207413056-03a22262-287a-4bc1-843e-4c5d3760b5b9.png">

If I tried to login, it will fail as my personal account cannot be used for Open Source Enterprise Cloud Access (1ES)
<img width="598" alt="image" src="https://user-images.githubusercontent.com/40202887/207414052-e0b15ba3-9ee7-485a-9eb0-264598bb92fd.png">

Expected behavior:

I expect to see the public issues / PRs even if not authenticated with my @microsoft account. 

Options:
 1. be able to access them directly and require another authentication only if doing something that requires more than public access (e.g. Approve some PR)
 
 2. Besides, "continue" in Single sign-on to Microsoft Open Source to have an option to "Continue as public".
 
 Workaround:
 
 I can open any issue in another browser where I'm not logged or just open it in Incognito and I can see them, but it's not really convenient.

**Comment by theficus** [comment]
I'm piling on to this in the hope that it gets some more visibility. I've also seen this mentioned in #9045 and #13088 but nothing has had any response as of yet.

This issue makes GitHub mostly unusable for me in any personal capacity. Even if I'm not doing things in a work capacity, I often have a need to view code, PRs, or issues in otherwise public Microsoft repositories. The SSO device management requirements make it so that I actually have fewer privileges against a public Microsoft repository than an _unauthenticated_ user! That feels deeply wrong to me.

There really needs to be a way to bypass this requirement for performing non-privileged operations against public repositories. I like the suggestion 2 as a way of being able to tell GitHub "just treat me like any other user out there rather than someone tied to an organization". As things are today, it's just not viable for me to have a single account on GitHub for personal and work things because of these restrictions.

**Reply by bryan-bar** [reply]
I am part of an organization and experiencing the same issue when trying to access any of the public repos on personal devices.
I do not want to use single sign-on as I do want to see private repos on personal devices.

I am unable to do the following without my organizations single-sign on:

- access organizations github page to see list of public repos 
- access public pull requests
- switch branches on the organization's public repo
- see contributions made to organization's public repo on my account page

I am still able to access the repo by searching for it, or with the link, as well as my own forks and branches.

If I am not logged in, I am able to access my organization's page, public repos, switch branches, read PRs, see contributions made to organizations public repos on my account.

**Comment by stefanflegg** [comment]
I also experience this problem after linking my personal GH account to the MSFT org. Are there any updates to share on this?

There are some other discussions on the same topic:
https://github.com/orgs/community/discussions/9045
https://github.com/orgs/community/discussions/13088

**Comment by qiyuangong** [comment]
This issue is blocking me from accessing the public repo of the organization on my laptop.

Unfortunately, it's designed to act this way. Found this description in doc https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-saml-single-sign-on-for-your-organization/about-identity-and-access-management-with-saml-single-sign-on#about-saml-sso

> IdP authentication is not required for accessing public repositories in certain ways:

> Viewing the repository's overview page and file contents on GitHub
Forking the repository
Performing read operations via Git, such as cloning the repository
**Authentication is required for other access to public repositories, such as viewing issues, pull requests, projects, and releases.**

> Note: SAML authentication is not required for outside collaborators. For more information about outside collaborators, see "[Roles in an organization](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization#outside-collaborators)."

Any idea of disabling it?

**Comment by David-Noble-at-work** [comment]
And I’ll add-on: I’m a former MS employee decoupled from MS having taken care to unlink my MS e-mail from my account. What more must I do to completely sever my relationship and join the good people of the open source community?

**Reply by jeffwilcox** [reply]
David, is this still an issue for you? On the Microsoft side we have taken a number of changes to work around this problem over the last year...

**Comment by devhl-labs** [comment]
I have the same issue. Its really frustrating that my personal account cant access things that an unauthenticated user can. If im looking at a public repo, why is my personal account not good enough? I left the microsoft org (within github), hopefully this has no ill effects, but it fixed my personal account.

**Reply by sandrohanea** [reply]
Just did the same, it looks like they are not going to fix this annoying issue so this can be considered a workaround.

 1. Created a new "personal" GitHub account only for MSFT orgs. (e.g. https://github.com/sandrohanea-msft)
 2. Left all Microsoft orgs from my personal account ([listed here](https://repos.opensource.microsoft.com/orgs))
 3. Unlinked my personal account from [here](https://repos.opensource.microsoft.com/link)
 4. Linked the new "personal" microsoft-only account (same link as above)
 5. Rejoined the orgs on the new account. 
 
 Now I'm able to see public stuff from my *real personal* account and the issue was just transferred to the other "personal" account which I'm using only for MSFT contributions or private repos.
 
 Apparently, VS supports 2 GitHub accounts and copilot, and everything still works (the reason why I was hesitant to do it initially):
 
![image](https://github.com/user-attachments/assets/8f84419c-fdd1-4818-ac94-6e3c40b4f30b)

**Comment by pinkfloydx33** [comment]
I only have a personal github account. It has never been linked to an organization of any sort, including Microsoft. Not being able to view code, access releases is annoying, but the inability to post an issue is also a problem. 

Oddly enough, in my profile it lists "[Microsoft Open Source](https://github.com/enterprises/microsoftopensource) Unaffiliated member" under the "Enterprises" section. What gives?

Is there some way around this without having to log out completely?

**Reply by Ryan-Palmer** [reply]
I had a pr merged into the webjobs sdk repo yesterday and now the same thing has happened to me. I'm not part of any MS org.

**Comment by amithkk** [comment]
This is happening to me as well, and I seem to be added to the same "enterprise" as @pinkfloydx33 above:
<img width="1084" height="201" alt="image" src="https://github.com/user-attachments/assets/dea7c47d-7d6a-454a-bb0b-fbab6adbaef6" />

I can no longer access the issues for any microsoft repo while being logged in (and of course, can't post new issues or even respond to comments on my own issues). I am assuming this is a problem for anyone who's ever contributed to a MS OSS repo

**Comment by pinkfloydx33** [comment]
I recieved an email two days ago that said I was removed from the organization. I can now access the repositories that I previously could not.

Thanks to whomever made that happen!

**Reply by Ryan-Palmer** [reply]
The Microsoft dev that merged my PR kindly got me removed and I shared this thread with him so maybe everyone got fixed? 🤞

**Reply by jeffwilcox** [reply]
Thankfully GitHub has shipped a setting allow this "unaffiliated" concept to be configured differently. https://github.blog/changelog/2025-10-28-unaffiliated-users-for-ghec-accounts-now-generally-available/
